src/Drupal/Commands/core/LocaleCommands.php

```diff
@@ -144,76 +145,121 @@ public function update($options = ['langcodes' => self::REQ])
      *   Export the Dutch non-customized translations.
      * @usage drush locale:export --langcode=nl --types=customized,not-customized > nl.po
      *   Export the Dutch customized and not customized translations.
-     * @usage drush locale:export > drupal.pot
+     * @usage drush locale:export --template > drupal.pot
      *   Export the template file to translate.
      * @aliases locale-export
      * @see \Drupal\locale\Form\ExportForm::submitForm
      *
      * @throws \Exception
      */
-    public function export($options = ['langcode' => self::OPT, 'types' => self::OPT])
+    public function export($options = ['template' => false, 'langcode' => self::OPT, 'types' => self::OPT])
     {
+        $language = null;
+        $poreader_options = null;
+        $template = (bool)$options['template'];
+
         // If template is required, language code is not given.
-        if ($options['langcode'] != LanguageInterface::LANGCODE_SYSTEM) {
-            $language = \Drupal::languageManager()->getLanguage($options['langcode']);
-
-            if ($language == null) {
-                $this->logger()->warning(dt('Language code @langcode is not configured.', [
-                    '@langcode' => $options['langcode'],
-                ]));
-                return;
-            }
-        } else {
-            $language = null;
+        if ($options['langcode'] != null) {
+            $language = $this->getLanguage($options['langcode']);
+        } elseif ($template === false) {
+            throw new \Exception(dt('Set --langcode or --template, see help for more information.'));
```

Comment:
Does Drush have exception types? If so, can we use it instead of the generic `\Exception`

Reply:
We have put off making specific exception types until we have a need to catch exceptions based on the type that is thrown. So far we have not had that need.

In this specific case, I would be -1 on making a Locale-specific exception to throw here. If we were going to use something other than \Exception, I'd want something like CommandFailedException, as the intention here is to stop the execution of the command and display an error message (with corresponding failure code) to the user. If something like that were to be defined, we'd probably want to put it in AnnotatedCommand.

If there are other exceptions that might be caught separately, then it would be okay to start making specific types. However, I think in most cases, a command implementation is only going to want to throw CommandFailedExceptions.